packages/ra-ui-materialui/src/field/DateField.tsx

```diff
@@ -61,7 +61,18 @@ const DateFieldImpl = <
     }
 
     const value = get(record, source);
-    if (value == null || value === '') {
+    const parsedDate = (value: any) => {
```

Comment:
I doin't understand why you extracted to a function just to call it afterwards.

Reply:
I did it to encapsulate the logic for validating whether the value can represent a Data or not. Probably a matter of style and personal preference, I can change it to inline code obviously.

Reply:
yes, please inline this code